README.md

```diff
@@ -158,6 +158,9 @@ require("oil").setup({
   trash_command = "trash-put",
   -- Selecting a new/moved/renamed file or directory will prompt you to save changes first
   prompt_save_on_select_new_entry = true,
+  -- The time in milliseconds it takes for oil buffers to close when no oil buffers are visible
+  -- Any number <= 0 or any non-number will make the buffers close instantly
```

Comment:
Oh, also small note that you only have to add this value and the comment in `config.lua`. A github action will ensure that it is copied here to the README and the vim help docs. Of course, it's also fine if you do this yourself.

Reply:
I personally dislike the idea of having a 1 be a "near-instant" option, because calling `vim.defer_fn` with a 1 millisecond delay does not ensure that the buffers will be cleared deleted after 1 millisecond, it only ensures that they will be deleted after *at least* 1 millisecond. Depending on other scheduled functions this might give enough time for users to jump back on accident.

So how about this:
- `0` disables the cleanup like you proposed (I really like that idea)
- `false` cleans up buffers instantly (technically that's already how it works but I would make this explicit in the documentation)

Reply:
I also added a small note about when the cleanup occurs, because I was worried users might think that deleting buffers instantly means we delete previous buffers while they are still inside an oil buffer.

I am open for rephrasing any part of the documentation if you have some improvements

Reply:
I would make the case that one magic value (`0`) is better than having two magic values of different types (`0` and `false`). Two values complicates the config API, complicates the code (now you have to add type checks), and requires additional explanation. Ostensibly the `false` value is being added for a feature that you won't be using, and I won't be using, so we're not sure that anyone even wants it.

Given that `cleanup_delay_ms = false` isn't a feature anyone needs right now, that it would be easy to add later, and that it complicates both the API and the code, I'd prefer if we leave it out.

The explanation on when buffers get cleaned up is 👍👍

Reply:
So if I understand correctly, you would want to keep `0` as the instant delete option?

I have to say I personally prefer the way it is now, mostly because it gives users more options.

A lot of plugins I use have these magic values, for example I recently switched to `lazy.nvim` and they use that a lot (see `config` for example), but generally I find them very nice to use. 

As for the type check, we have to do that anyways (I also did that in the previous version of the code) because there is nothing enforcing that the user actually passes a number and I think that a wrong config shouldn't create weird errors like `bad argument #1 to 'start'` (which is what you get when passing another type as delay to `defer_fn`)

That being said, it is your call and I'm fine with both, just wanted to share my personal opinion.

Reply:
To clarify, I think there _should_ be a value that can disable the cleanup timeout entirely, and there _should not_ be a value that cleans up buffers "instantly". I would be fine if `cleanup_delay_ms = 0` disables the cleanup, or if it's `cleanup_delay_ms = false`. Those both seem reasonable.

I don't have a problem with magic values when they're needed (though I do prefer to avoid them when possible), I have a problem with API complexity. I'm splitting hairs over this because with one of my earlier plugins I didn't push back on feature requests or PRs and as a consequence the config options became a massive garbage bin, even after two large refactors. Once you add something, it take a long time and a lot of effort to remove it, so I prefer to only add options that are necessary, not "well I could see how someone might want this in the future". If someone does want it in the future, they can ask or make a PR. If (more likely) no one does, then we've saved ourselves from needless complications.

As a further explanation for why the "instant cleanup" isn't needed, with a delay of 1ms even if the scheduler doesn't run it after exactly 1ms, it's not going to add any perceptible delay to any human. The only case where it could matter is if you have autocmds or other scheduled functions that rely on the oil buffers getting cleaned up. I would say that first, don't do that it's a bad idea. But also, that's never going to work because if you look at the surrounding code there's an initial delay of 10ms already. That delay is there because switching buffers can sometimes be accompanied by autocmds that do some further reshuffling of the buffers/windows, and we want to let all of that settle before checking if the oil buffers can be cleaned. Adding another 1ms defer on top of that isn't going to change anything.

Reply:
I guess this sentence here
> Ostensibly the false value is being added for a feature that you won't be using, and I won't be using, so we're not sure that anyone even wants it

confused me since the feature you are referring to is *literally* the feature that I created this PR for.

I can't help but feel a bit disappointed, since setting the delay to `1` to make it "basically instant" feels like a hack to me and I really like making things absolute. But I understand where you're coming from and ultimately I trust in your judgement and your experience.

I will make the changes necessary ASAP :+1:

Reply:
Apologies, I misread your original comment. I that case...I think I would be fine to go with your first suggestion, with the values flipped

- `cleanup_delay_ms = false` disables the cleanup entirely
- `cleanup_delay_ms = 0` does the immediate cleanup

With this setup, I don't think 0 has to be called out as a magic value because it simply functions as you would expect.

Reply:
Damn.. I feel a bit stupid for not implementing it that way around in the first place, it makes a lot more sense.

Anyway great idea, I implemented it now so please check the code again ^^